src/app/utils/pickupTimeEstimator.js

```diff
@@ -107,28 +104,74 @@ estimator._addOnsiteTravelDuration = async (serviceTime, locationId) => {
   return estimator._addDuration(serviceTime, onsiteTravelDuration)
 }
 
-
 /**
- *  Returns true if the named location has a special delivery schedule (e.g. is
- *  a scholar room)
- *
- *  TODO: Implement
- */
-estimator._hasSpecialDeliverySchedule = (locationId) => {
-  // Return true if locationId matches room known to have a specific delivery schedule
-  return false
-}
-
-/**
- *  Given a location id and a timestamp string, returns a new timestamp string
- *  representing the future delivery time for the location given the known
- *  delivery schedule.
- *
- *  TODO: Implement
+ * Given a location id and a timestamp string, returns an object with:
+ * hasSpecialDeliverySchedule: a Boolean which is true if the location matches a
+ * location with a special delivery schedule
+ * adjustedSpecialScheduleTime: a timestamp string, representing the future delivery time for
+ * the given known delivery schedule
  */
 estimator._adjustToSpecialSchedule = (locationId, time) => {
-  // Update time to next delivery schedule after given time
-  return time
+  let hasSpecialDeliverySchedule = false
+  let adjustedSpecialScheduleTime = new Date(time)
+  let secondFloorScholarRooms = ['mal17', 'mala', 'malc', 'maln', 'malw']
+  let mapRooms = ['mapp8', 'mapp9', 'map08']
+  let firstHour
+  let getNextHour
+  let lastHour
+
+  if (secondFloorScholarRooms.includes(locationId)) {
+    hasSpecialDeliverySchedule = true
+    firstHour = 10
+    lastHour = 16
```

Comment:
`firstHour` and `lastHour` should be bound by the actual service hours for the day. A Mon 5pm closure means last delivery is 4. I would rely on [_getNextServiceHours](https://github.com/NYPL/discovery-front-end/blob/15c3dd8ca7d2a88e4c40d338b65bcc30ef763100/src/app/utils/pickupTimeEstimator.js#L319) to retrieve the real window you need to work with, and constrain hours to that. Something like:
```
if (secondFloorScholarRooms.includes(locationId)) {
    hasSpecialDeliverySchedule = true
    const serviceHours = await estimator._getNextServiceHours(locationId, time)
    firstHour = // ensure this is an even number greater than the hour value in serviceHours.startTime (accounting for NY offset)
    lastHour = // ensure this is an even number less than the hour value in serviceHours.endTime (accounting for NY offset)
```